index.js

```diff
@@ -0,0 +1,14 @@
+const validatePassword = (valid) => {
+
+  const complexityCheck = /(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()-+])/
```

Comment:
The problem with this is that it pushed the bounds of readability and definitely makes maintaining it harder. If the rules change it's tougher to update. Even breaking it down into individual checks would be a better solution.

```JavaScript
const hasLowerCase = /(?=.*[a-z])/
const hasUpperCase = /(?=.*[A-Z])/
```